jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js

```diff
@@ -205,6 +205,19 @@ var module = module;
         // if the macaroon is valid.
         false
       );
+    },
+
+    /**
+      Lists all the templates for cloud credentials for a specific user.
+
+      @method listTemplates
+      @param callback {Function} A callback to handle errors or accept the data
+          from the request. Must accept an error message or null as its first
+          parameter and the response data as its second.
+    */
+    listTemplates: function(callback) {
+      var url = this.jemURL + '/template';
+      return _makeRequest(this.bakery, url, 'GET', null, callback, true);
```

Comment:
I don't think we want to massage too much, if only b/c then we have different expectations for what each method does; however, like the listServers method this should probably return the data inside the templates key, rather than the whole response. This will also involve a test update.